app/src/main/java/com/sdp/movemeet/models/Activity.java

```diff
@@ -84,7 +84,7 @@ public Activity(String activityId,
         this.longitude = longitude;
         this.latitude = latitude;
         this.description = description;
-        this.documentPath = documentPath;
+        super.documentPath = documentPath;
```

Comment:
Why did you use "super" instead of "this" here ? documentPath is already in the constructor ?

Reply:
Since `FirebaseObject` is now an abstract class, it has its own `documentPath` attribute which it uses in its default implementation of `getDocumentPath()` and `setDocumentPath()`. Here, I'm setting the document path in the superclass so that the abstract functions can run correctly.

Reply:
Okay I see, `Activity.java` is in fact inheriting from (i.e. implementing) `FirebaseObject`!